samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -125,6 +125,7 @@ Config generateConfig() {
     if (isIntermediate()) {
       config.put(String.format(StreamConfig.IS_INTERMEDIATE_FOR_STREAM_ID(), spec.getId()), "true");
       config.put(String.format(StreamConfig.CONSUMER_OFFSET_DEFAULT_FOR_STREAM_ID(), spec.getId()), "oldest");
+      config.put(String.format(StreamConfig.PRIORITY_FOR_STREAM_ID(), spec.getId()), "1");
```

Comment:
If a user already has prioritized streams, would this put the intermediate stream at the same priority as their input?. Alternately, You could set to INT_MAX here to avoid conflicting with configured priorities?

Reply:
Makes sense, will change to INT_MAX